modules/shortcodes/js/slideshow-shortcode.js

```diff
@@ -143,6 +143,23 @@ JetpackSlideshow.prototype.renderControls_ = function() {
 		var controlName = controls[ i ];
 		var a = document.createElement( 'a' );
 		a.href = '#';
+		switch ( i ) {
+			case 0:
+				a.className = 'classic-jetpack-slideshow_button-previous';
+				a.setAttribute( 'aria-label', jetpackSlideshowSettings.label_prev );
+				break;
+
+			case 1:
+				a.className = 'classic-jetpack-slideshow_button-pause';
+				a.setAttribute( 'aria-label', jetpackSlideshowSettings.label_pause );
+				break;
+
+			case 2:
+				a.className = 'classic-jetpack-slideshow_button-next';
+				a.setAttribute( 'aria-label', jetpackSlideshowSettings.label_next );
+				break;
+		}
+		a.setAttribute( 'role', 'button' );
```

Comment:
I am thinking we may be able to simplify this a bit if we stick to the existing control names (`'prev', 'stop', 'next'`) instead of `prev, pause, next`. Here is how I'd do it for example:

```diff
diff --git a/modules/shortcodes/js/slideshow-shortcode.js b/modules/shortcodes/js/slideshow-shortcode.js
index 27ebd8d4a..f90456818 100644
--- a/modules/shortcodes/js/slideshow-shortcode.js
+++ b/modules/shortcodes/js/slideshow-shortcode.js
@@ -141,25 +141,14 @@ JetpackSlideshow.prototype.renderControls_ = function() {
 	var controls = [ 'prev', 'stop', 'next' ];
 	for ( var i = 0; i < controls.length; i++ ) {
 		var controlName = controls[ i ];
+		var label_name = 'label_' + controlName;
 		var a = document.createElement( 'a' );
+
 		a.href = '#';
-		switch ( i ) {
-			case 0:
-				a.className = 'classic-jetpack-slideshow_button-previous';
-				a.setAttribute( 'aria-label', jetpackSlideshowSettings.label_prev );
-				break;
-
-			case 1:
-				a.className = 'classic-jetpack-slideshow_button-pause';
-				a.setAttribute( 'aria-label', jetpackSlideshowSettings.label_pause );
-				break;
-
-			case 2:
-				a.className = 'classic-jetpack-slideshow_button-next';
-				a.setAttribute( 'aria-label', jetpackSlideshowSettings.label_next );
-				break;
-		}
+		a.className = 'button-' + controlName;
+		a.setAttribute( 'aria-label', jetpackSlideshowSettings[label_name] );
 		a.setAttribute( 'role', 'button' );
+
 		controlsDiv.appendChild( a );
 		this.controls[ controlName ] = a;
 	}
diff --git a/modules/shortcodes/slideshow.php b/modules/shortcodes/slideshow.php
index 045d96f3b..ae0aadbdd 100644
--- a/modules/shortcodes/slideshow.php
+++ b/modules/shortcodes/slideshow.php
@@ -281,11 +281,8 @@ class Jetpack_Slideshow_Shortcode {
 			 * @param array $args
 			 * - string - spinner        - URL of the spinner image.
 			 * - string - speed          - Speed of the slideshow. Defaults to 4000.
-			 * - string - class_prev     - CSS class for slideshow's previous button
-			 * - string - class_pause    - CSS class for slideshow's pause button
-			 * - string - class_next     - CSS class for slideshow's next button
 			 * - string - label_prev     - Aria label for slideshow's previous button
-			 * - string - label_pause    - Aria label for slideshow's pause button
+			 * - string - label_stop     - Aria label for slideshow's pause button
 			 * - string - label_next     - Aria label for slideshow's next button
 			 */
 			apply_filters(
@@ -293,9 +290,9 @@ class Jetpack_Slideshow_Shortcode {
 				array(
 					'spinner'     => plugins_url( '/img/slideshow-loader.gif', __FILE__ ),
 					'speed'       => '4000',
-					'label_prev'  => __( 'Previous Slide', 'jetpack' ),
-					'label_pause' => __( 'Pause Slideshow', 'jetpack' ),
-					'label_next'  => __( 'Next Slide', 'jetpack' ),
+					'label_prev' => __( 'Previous Slide', 'jetpack' ),
+					'label_stop' => __( 'Pause Slideshow', 'jetpack' ),
+					'label_next' => __( 'Next Slide', 'jetpack' ),
 				)
 			)
 		);

```

Reply:
This is brilliant. I didn't realise I'm using switch in a for loop, your approach is much better. I'll do that. Actually I have done it but haven't tested it because I ran into some issues with my dev environment. I'll test it soon and request a review.